From 9b891c10e3be08b7f0c8edfbf1cd9a893e6db8ea Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Wed, 22 Jul 2026 21:04:30 -0400
Subject: [PATCH 216/268] atiadlxx: Add semi-stub for
 ADL2_Graphics_VersionsX3_Get.

Used by AMD AGS.
---
 dlls/atiadlxx/atiadlxx.spec   | 1 +
 dlls/atiadlxx/atiadlxx_main.c | 7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/dlls/atiadlxx/atiadlxx.spec b/dlls/atiadlxx/atiadlxx.spec
index 3e7c7ac19c4..2b916b922c1 100644
--- a/dlls/atiadlxx/atiadlxx.spec
+++ b/dlls/atiadlxx/atiadlxx.spec
@@ -433,6 +433,7 @@
 @ stub ADL2_Graphics_MantleVersion_Get
 @ cdecl ADL2_Graphics_Platform_Get(ptr ptr)
 @ cdecl ADL2_Graphics_VersionsX2_Get(ptr ptr)
+@ cdecl ADL2_Graphics_VersionsX3_Get(ptr long ptr)
 @ cdecl ADL2_Graphics_Versions_Get(ptr ptr)
 @ stub ADL2_Graphics_VulkanVersion_Get
 @ stub ADL2_HybridGraphicsGPU_Set
diff --git a/dlls/atiadlxx/atiadlxx_main.c b/dlls/atiadlxx/atiadlxx_main.c
index 492d7e2030a..4d732860414 100644
--- a/dlls/atiadlxx/atiadlxx_main.c
+++ b/dlls/atiadlxx/atiadlxx_main.c
@@ -700,6 +700,13 @@ int CDECL ADL2_Adapter_NumberOfAdapters_Get(ADL_CONTEXT_HANDLE ctx, int *count)
     return ADL_OK;
 }
 
+int CDECL ADL2_Graphics_VersionsX3_Get(ADL_CONTEXT_HANDLE ptr, int adapter_index, ADLVersionsInfoX2 *ver)
+{
+    FIXME("ptr %p, index %d, ver %p semi-stub.\n", ptr, adapter_index, ver);
+    memcpy(ver, &version2, sizeof(version2));
+    return ADL_OK;
+}
+
 int CDECL ADL2_Graphics_VersionsX2_Get(ADL_CONTEXT_HANDLE ptr, ADLVersionsInfoX2 *ver)
 {
     TRACE("ptr %p, ver %p.\n", ptr, ver);
-- 
2.55.0

